File: Java_Generated/Java/Maven/src/main/java/measurements/RandomGenerator.java
package measurements;

import diagram.interfaces.IDesktop;
import measurements.interfaces.IMeasurement;

import java.util.Random;

public class RandomGenerator  extends  Measurements implements IMeasurement {
    public RandomGenerator(String name, IDesktop desktop) {
        super(name, desktop);
        addMeasurement(this);
    }

    @Override
    public String getMeasurementName() {
        return "Random";
    }

    @Override
    public Object getMeasurementType() {
        return type;
    }

    @Override
    public Object getMeasurementValue() {
        return value;
    }

    @Override
    public String toString()
    {
        return value[0] + "";
    }

    @Override
    public void updateMeasurements() {
        value[0] = random.nextDouble();
    }
    double[] type = new double[0];
    double[] value = new double[]{0};

    Random random = new Random();

}

File: src/Python/lib/measurements/random_generator.py
import random
from ..measurements.measurements import Measurements
from measurement import Measurement
from ..desktop import Desktop

class RandomGenerator(Measurements, Measurement):
    def __init__(self, name: str, desktop: Desktop, seed=-1):
        Measurements.__init__(self, name=name, desktop=desktop)
        Measurement.__init__(self, name=self.name, type=float)
        self.append(self)
        random.seed(seed)
        self.random = random.random
        
        self.value = 0
    
    def measurement_name(self):
        return "Random"
    
    def measurement_type(self):
        return float
    
    def measurement_value(self):
        return self.measurement_type()
    
    def __str__(self):
        return self.value
    
    def update(self):
        self.value = self.random()
